docs/src/resources/troubleshooting.md

```diff
@@ -1,5 +1,101 @@
 # Troubleshooting
 
-This guide helps you troubleshoot common issues with Runwasi.
+This guide helps you troubleshoot common issues with runwasi shims.
 
-[TODO]
+## Container Fails to Start
+
+If your wasm container fails to start, you might see events like these:
+
+```
+Events:
+  Type     Reason                  Age               From               Message
+  ----     ------                  ----              ----               -------
+  Normal   Scheduled               18s               default-scheduler  Successfully assigned default/wasi-demo-75d6bb666c-479km to minikube
+  Warning  FailedCreatePodSandBox  4s (x2 over 18s)  kubelet            Failed to create pod sandbox: rpc error: code = Unknown desc = failed to get sandbox runtime: no runtime for "wasmtime" is configured
+```
+
+#### Investigation Steps:
+
+1. **Check runtime class configuration**: Ensure your `RuntimeClass` is correctly configured.
+   ```bash
+   kubectl get runtimeclass -o yaml
+   ```
+
+2. **Check containerd configuration**: Verify that the containerd configuration has a runtime configured for your Wasm shim (e.g., wasmtime).
+   ```bash
+   cat /etc/containerd/config.toml
+   
+   # Look for a section like:
+   [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.wasmtime]
+      runtime_type = "io.containerd.wasmtime.v1"
+   ```
+
+3. Make sure the containerd shim binary is installed properly.
+
+## Log-Based Troubleshooting
+
+Examining logs is one of the most effective ways to troubleshoot issues in runwasi shims.
+
+### Finding containerd Logs
+
+Logs can be in different locations depending on your system:
+
+#### Kubernetes Distributions
+- k3s: `/var/lib/rancher/k3s/agent/containerd/containerd.log`
+- k8s (kubeadm): `/var/log/containerd/containerd.log`
+
+#### Using journalctl (systemd-based systems)
+```bash
+# View all containerd logs
+journalctl -u containerd
+
+# Follow containerd logs
+journalctl -u containerd -f
+```
+
+### Kubernetes Pod Logs
+
+In addition to containerd logs, Kubernetes pod logs show the logs from your containers.
+
+```bash
+kubectl logs <pod-name>
+
+# Or stream logs in real-time
+kubectl logs -f <pod-name>
+```
+
+### Using Structured Logging with Container and Pod IDs
+
+runwasi shims include structured logging that automatically adds container ID and pod ID to log messages, making it easier to filter logs.
+
+#### Finding Container ID and Pod ID:
+
+```bash
+# Get container IDs for a pod
+kubectl get pod <pod-name> -o jsonpath='{.status.containerStatuses[*].containerID}'
```

Comment:
I believe the pod id is in the output too, most folks won't have crictl

Reply:
emm I wasn't able to find the pod ID using kubectl. Do you know the command to do that?

Reply:
I thought there was a simple way to do this but I guess there isn't. `critctl` or finding the container id (which will have it on there will work). Thanks for looking into this